src/Providers/BlogCoreServiceProvider.php

```diff
@@ -94,7 +94,17 @@ protected function registerConfigs(): void
 
         foreach ($files as $file) {
             if (pathinfo($file, PATHINFO_EXTENSION) === 'php') {
-                $this->mergeConfigFrom($configDir . '/' . $file, pathinfo($file, PATHINFO_FILENAME));
+                $configName = pathinfo($file, PATHINFO_FILENAME);
+                $configPath = $configDir.'/'.$file;
+
+                if (file_exists(config_path($configName.'.php'))) {
```

Comment:
The updated code adds a check to see if the configuration file exists before merging. If it does, the new configuration is merged with the existing one using `config()->set` and `array_merge`. If the file doesn’t exist, it falls back to the original method of merging with `mergeConfigFrom`. This change enhances flexibility in handling configuration files.